var/spack/repos/builtin/packages/bufr-query/package.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2013-2024 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class BufrQuery(CMakePackage):
+    """The NOAA bufr-query Library can be used to read NCEP and WMO formated BUFR
+    files using a simple interface that does not require the user to know the
+    details of the BUFR format. Detailed documentation for the BUFR Library can
+    be found at https://bufr-query.readthedocs.io/en/latest/index.html"""
+
+    homepage = "https://github.com/NOAA-EMC/bufr-query"
+    url = "https://github.com/NOAA-EMC/bufr-query/archive/refs/tags/v0.0.1.tar.gz"
+    maintainers("srherbener", "rmclaren")
+
+    license("Apache-2.0", checked_by="srherbener")
+
+    version("0.0.2", sha256="b87a128246e79e3c76e3158d89823e2ae38e9ee1a5a81b6f7b423837bdb93a1f")
+    version("0.0.1", sha256="001990d864533c101b93d1c351edf50cf8b5ccc575e442d174735f6c332d3d03")
+
+    # Required dependencies
+    depends_on("ecbuild", type=("build"))
+    depends_on("llvm-openmp", when="%apple-clang", type=("build", "run"))
+    depends_on("mpi", type=("build", "run"))
+    depends_on("eckit@1.24.4:", type=("build", "run"))
+    depends_on("eigen@3:", type=("build", "run"))
+    depends_on("gsl-lite", type=("build", "run"))
+    depends_on("netcdf-c", type=("build", "run"))
+    depends_on("netcdf-cxx4", type=("build", "run"))
+    depends_on("bufr", type=("build", "run"))
+
+    # Optional dependencies
+    variant("python", default=True, description="Enable Python interface")
+    depends_on("python@3:", type=("build", "run"), when="+python")
+    depends_on("py-pybind11", type=("build"), when="+python")
+
+    # CMake configuration
+    def cmake_args(self):
+        args = [self.define_from_variant("BUILD_PYTHON_BINDINGS", "python")]
+
+        # provide path to netcdf-c include files
+        nc_include_dir = Executable("nc-config")("--includedir", output=str).strip()
+        args.append("-DCMAKE_C_FLAGS=-I" + nc_include_dir)
+        args.append("-DCMAKE_CXX_FLAGS=-I" + nc_include_dir)
```

Comment:
Why is that needed? If bufr-query is a `cmake` package, it should run `find_package(netcdf)` itself, or?

Reply:
This is what I found in other formulas to fix the problem where the netcdf.h file cannot be found. Here is the find_package in the bufr_query CMakeLists.txt.

https://github.com/NOAA-EMC/bufr-query/blob/a78d4c79d72aef3a7046386090ee5c3c51554718/CMakeLists.txt#L30

Im guessing that asking for only the CXX component of the netcdf library is the issue. However, it seems that asking for only the CXX component should realize that you have to also use the C component and the netcdf find package process would bring that in for you. It must be attempting to do that but the include path is not getting set - once that is done it finds the libraries just fine.

I can verify if this truly is the issue.

What's the best way to proceed?
- I could replace this code with a patch that adds the C component to the bufr_query netcdf find_package call
- OR We could try to get the fix into bufr_query and remove this code
- OR leave it as is

Thanks!

Reply:
Adding the C component in the netcdf find_package call in the bufr-query CMakeLists.txt did not help - still got the cannot file "netcdf.h" include file. But we can leave as is for now (as noted above).